src/c++/perf_analyzer/mock_data_loader.h

```diff
@@ -35,16 +35,29 @@ namespace triton { namespace perfanalyzer {
 ///
 class MockDataLoader : public DataLoader {
  public:
-  MockDataLoader()
+  MockDataLoader() { SetupMocks(); }
+  MockDataLoader(size_t batch_size) : DataLoader(batch_size) { SetupMocks(); }
+
+  void SetupMocks()
   {
     ON_CALL(*this, GetTotalSteps(testing::_))
         .WillByDefault([this](size_t stream_id) -> size_t {
           return this->DataLoader::GetTotalSteps(stream_id);
         });
+    EXPECT_CALL(*this, GetTotalSteps(testing::_)).Times(testing::AnyNumber());
   }
 
   MOCK_METHOD(size_t, GetTotalSteps, (size_t), (override));
 
+
+  cb::Error ReadDataFromJSON(
```

Comment:
Wanna use gmock to accomplish the same thing since we're starting to use the mocking framework in more places, now?:

```cpp
class MockDataLoader : public DataLoader {
...
  void SetupMocks()
  {
...
    ON_CALL(*this, ReadDataFromJSON(testing::_, testing::_, testing::_))
        .WillByDefault(
            [this](
                const std::shared_ptr<ModelTensorMap>& inputs,
                const std::shared_ptr<ModelTensorMap>& outputs,
                const std::string& json_file) -> cb::Error {
              return this->ReadDataFromStr(json_file, inputs, outputs);
            });
  }
...
  MOCK_METHOD(
      cb::Error, ReadDataFromJSON,
      (const std::shared_ptr<ModelTensorMap>&,
       const std::shared_ptr<ModelTensorMap>&, const std::string& json_file),
      (override));
...
};
```

Reply:
Can you explain the benefit of writing all of that just to do a simple override?  It seems like the value is the ability for the test to control whether it does the new behavior or the old behavior. Do we need that? Is there another reason to use gmock for this?

Reply:
The only reason I can think of is consistency.